### ISSUE TITLE
Memory Leak when reload a new project or click File->New

### ISSUE THREAD

**Issue by altjz** [LLK/scratch-gui#6309]
### Expected Behavior

Should release the memory(Including GPU Memory) when click File->New Or reload a new project

### Actual Behavior
1. before load any project
<img width="480" alt="before" src="https://user-images.githubusercontent.com/10369139/97386825-aea22d80-190f-11eb-9dbf-1a1e502e0a90.png">
2. load a large project
<img width="480" alt="after" src="https://user-images.githubusercontent.com/10369139/97386833-b366e180-190f-11eb-8ec4-954a89229ef5.png">
3. after click File->New
<img width="485" alt="clicknew" src="https://user-images.githubusercontent.com/10369139/97386850-bb268600-190f-11eb-9e19-dcc3647e6998.png">

You can see the Memory usage only reduce 5%,  GPU Memory usage not at all.

This will be a disaster on Scratch Desktop, you reload a project or create new A project, it won't release any memory until you restart the desktop.

### Steps to Reproduce

1. Open a large project like this [FIFA 19](https://scratch.mit.edu/projects/248674749/editor/)
2. Open the Task Manager (chrome)
2. Click File->New

### Operating System and Browser

Mac OS 10.15.6 Chrome 86
Same on Windows

Same issue on Scratch Desktop

**Comment by BryceLTaylor** [comment]
Can a user perceive this issue?  What do they have to do for this to be noticeable to the user?

**Comment by cwillisf** [comment]
There could be many reasons for this, so it might make sense to split this into multiple issues as we discover causes. We probably do have one or more resource leak(s), but in addition to that I suspect the browser is not very aggressive in decreasing the space allocated to JavaScript.

One thing to look into: at first glance it looks like we might not reliably unregister costumes from the renderer when a sprite or costume gets deleted. If so, we should call `runtime.renderer.destroySkin(...)` to clean up.

**Comment by cwillisf** [comment]
My guess is that most users won't see any effect from this, but some might see a slowdown of Scratch or their overall computer (swap thrashing). In the worst case I think it would lead to the browser tab or app crashing (or being terminated due to exceeding limits) which could lead to data loss, but I suspect that's very rare.

If we get more information that suggests this is worse then we should adjust the labels, but for now I think "low impact" and "medium severity" is a reasonable estimate.

**Comment by altjz** [comment]
> There could be many reasons for this, so it might make sense to split this into multiple issues as we discover causes. We probably do have one or more resource leak(s), but in addition to that I suspect the browser is not very aggressive in decreasing the space allocated to JavaScript.
> 
> One thing to look into: at first glance it looks like we might not reliably unregister costumes from the renderer when a sprite or costume gets deleted. If so, we should call `runtime.renderer.destroySkin(...)` to clean up.

I found this probably cause by the `renderer._allSkins`.

the below code would reduce 90% memory usage from last project. But I can't tell no other side effect behind this behavior.

Reproduce Step (suggest use incognito mode)
1. Open the official site: [https://scratch.mit.edu/projects/editor/](https://scratch.mit.edu/projects/editor/)
2. Take Snapshot on devtool (Memory Tab), probably around 30 - 40 MB.
3. Open a Large Project, (use File->Load from your computer) eg. [FIFA19](https://scratch.mit.edu/projects/248674749/editor/)
4. Take Snapshot on devtool, probably around 850MB (use the FIFA 19 project)
5. Click File -> New
6. Take Snapshot, only reduce 25 MB
7. Run the below code in console
```js
// Find React Instance
function getReactInstance(dom) {
    for (const k in dom) {
        if (k.startsWith('__reactInternalInstance$')) {
            return dom[k];
        }
    }
}
// Find VM
function findVm(reactInstance) {
    if (reactInstance.return) {
        if (reactInstance.return.stateNode) {
            if (reactInstance.return.stateNode.props && reactInstance.return.stateNode.props.vm) {
                return reactInstance.return.stateNode.props.vm;
            }
        }
        return findVm(reactInstance.return);
    }
}
let vm = findVm(getReactInstance(document.querySelector(".injectionDiv").parentElement));

// execute the dispose() for all Skins
vm.runtime.renderer._allSkins.forEach((skin) => {
      if (skin) {
        skin.dispose();
      }
    });
// clear the skin array
vm.runtime.renderer._allSkins = []
```
8. Click `Collect garbage` icon then take a snapshot
9. Reduce 90% memory usage

![after](https://user-images.githubusercontent.com/10369139/97675097-a4c42a00-1ac9-11eb-9477-ddeb9993fea1.png)

**Comment by apple502j** [comment]
@cwillisf Note that the changes above will be blocked by LLK/scratch-vm#2661 because that bug currently causes clones to keep using the deleted costume.
